packages/delivery-react-native/delivery.js

```diff
@@ -1,11 +1,20 @@
 module.exports = (client, NativeClient) => ({
   sendEvent: (payload, cb = () => {}) => {
     const event = payload.events[0]
-    // this is because JS beforeSend operates on the event – report is not in scope
-    // report.threads = event.get('threads')
-    NativeClient.dispatch(event.toJSON()).then(() => cb()).catch(cb)
+    NativeClient.dispatch({
+      errors: event.errors,
+      severity: event.severity,
+      severityReason: event._handledState.severityReason,
+      unhandled: event.unhandled,
+      app: event.app,
+      device: event.device,
+      threads: event.threads,
+      breadcrumbs: event.breadcrumbs,
+      context: event.context,
+      user: event._user,
+      metadata: event._metadata,
+      groupingHash: event.groupingHash
+    }).then(() => cb()).catch(cb)
   },
-  sendSession: () => {
-    // TODO: log/warn here? If this gets used the something has gone wrong…
-  }
+  sendSession: () => {}
```

Comment:
Going off the previous comments something it appears that this isn't expected to be called - it'd be good to add some sort of warning message here if this does occur.